### PR TITLE
Implement Ellipse API

### PR DESCRIPTION
**Pull request by joone**
Bug: https://github.com/servo/servo/issues/17598

The ellipse API was added as public API of Azure and the Arc method was updated to deal with the ellipse API.

---
This change is [Reviewable](https://reviewable.io/reviews/servo/rust-azure/272)

### CODE REVIEW

**Comment by joone** [comment]
How can we build a test case and run it?

**Comment by joone** [comment]
Here is another patch for Servo side: https://github.com/joone/servo/commit/427a24a2ae523a9704bc7560b05e551a886cd9ef

File: libazure/2D.h [hunk 1]
```diff
@@ -448,7 +448,8 @@ class PathSink : public RefCounted<PathSink>
    */
   virtual void Close() = 0;
   /** Add an arc to the current figure */
-  virtual void Arc(const Point &aOrigin, float aRadius, float aStartAngle,
+  virtual void Arc(const Point &aOrigin, float aRadiusX, float aRadiusY,
+                   float aRotationAngle, float aStartAngle,
```
Comment (emilio) [inline libazure/2D.h]: This looks like it should be upstreamed? i'm actually surprised moz2d didn't have a way to do this...

**Comment by jdm** [comment]
We will either need to upstream the changes and update from upstream once that happens, or add a local .patch file that contains the changes to libazure to make it easier to update from upstream later.

**Comment by jdm** [comment]
Note: upstream at https://hg.mozilla.org/mozilla-central/file/tip/gfx/2d

**Comment by joone** [comment]
jdm@ how can I add a local .patch file to libazure?

**Comment by jdm** [comment]
Create a file that contains the diff of the changes to libazure, and add it to a directory named `patches`

**Comment by joone** [comment]
@jdm I created a patch in my commit, but the patch was not applied to libazure during the build. Could you take a look at the commit?

**Comment by jdm** [comment]
Right, there's no automated way to apply patches. Your changes still need to be applied to the source files directly, as well. Adding the patch file means that we won't lose track of them and overwrite them whenever we update from upstream in the future.

**Comment by joone** [comment]
@jdm, okay, I've updated the patch. Thanks!

**Comment by jdm** [comment]
Sorry, one last change - can we move the patches directory out of libazure? Right now it's easy to end up overwriting it. Please update the version in Cargo.toml to be 0.21.0, too.

**Comment by joone** [comment]
@jdm, I've done with the update.

**Comment by jdm** [comment]
@bors-servo: r+
Thanks!

**Pull request merged**